SPEC.md

```diff
@@ -865,9 +865,80 @@ task literals_paths {
     # does not exist, then `f2` is set to `None`.
     File? f2 = "/foo/bar.txt"
   }
+
+  # If baz.txt does not exist, this is an error.
+  File f3 = "baz.txt"
+
+  # If qux.txt does not exist, this is set to `None`.
+  File? f4 = "qux.txt"
 }
 ```
 
+The interpretation of relative paths (paths that do not start with `/`) depends on the context in which they appear:
+
+* *Outside the `output` section (e.g., in `input` or private declarations)*, relative paths are interpreted relative to the parent directory of the WDL document itself on the host filesystem, similar to how [import](#import-statements) paths are resolved.
+* *Inside the `output` section*, relative paths are interpreted relative to the task's execution directory. This is where task commands create their output files. See [Task Outputs](#task-outputs) for details.
+
+In both contexts, if an optional `File?` or `Directory?` declaration refers to a path that does not exist, the value is set to `None`.
+
+Absolute paths (paths starting with `/`) refer to specific locations on the host filesystem when used outside the `output` section. Within the `output` section, absolute paths may be interpreted in a container-dependent way—see [Task Outputs](#task-outputs) for details.
+
+<details>
+<summary>
+Example: relative_paths_context.wdl
+
+```wdl
+version 1.3
+
+task relative_paths_context {
+  input {
+    # This relative path is resolved relative to the WDL document's parent directory.
+    File input_file = "hello.txt"
+  }
+
+  command <<<
+    cat ~{input_file} > output.txt
+  >>>
+
+  output {
+    # This relative path is resolved relative to the execution directory.
+    File result = "output.txt"
+    String content = read_string(result)
```

Comment:
@claymcleod can we add example of relative paths outside the output directory `/` and what the expected results of also missing files. I feel like this test doesn't really cover the update definition

Reply:
This test is showing the correct behavior and is meant to be an integration tests that succeeds. It covers the updated definition by ensuring that `hello.txt` must resolve relative to the WDL document's parent directory and printing the correct results. Though the semantics of a `File` in input blocks and in private declarations in the same, I can change it to a private declaration to make it a bit more clear.